### PR TITLE
add Book model, repository and service

### PR DESCRIPTION
**Pull request by hlibmurphy**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'adopt'
+          cache: maven
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots verify
```
Comment (okuzan) [inline .github/workflows/ci.yml:18]:
```suggestion
        run: mvn --batch-mode --update-snapshots verify

```

File: checkstyle.xml [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+        <module name="EmptyCatchBlock">
+            <property name="exceptionVariableName" value="expected"/>
+        </module>
+        <module name="CommentsIndentation"/>
+        <module name="DeclarationOrder"/>
+        <module name="SingleSpaceSeparator">
+            <property name="validateComments" value="true"/>
+        </module>
+    </module>
+</module>
```
Comment (okuzan) [inline checkstyle.xml:250]:
```suggestion
</module>

```

File: src/main/java/com/github/onlinebookstore/repositories/impl/BookRepositoryImpl.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Comment (okuzan) [inline src/main/java/com/github/onlinebookstore/repositories/impl/BookRepositoryImpl.java]: add implementation

File: src/main/java/com/github/onlinebookstore/services/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+@Service
+public class BookServiceImpl implements BookService {
+    @Override
+    public Book save(Book book) {
+        return null;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return null;
```
Comment (okuzan) [inline src/main/java/com/github/onlinebookstore/services/impl/BookServiceImpl.java]: add implementation

File: src/main/java/com/github/onlinebookstore/OnlineBookStoreApplication.java [hunk 5]
```diff
@@ -1,13 +1,31 @@
...
+    public CommandLineRunner commandLineRunner() {
+        return new CommandLineRunner() {
+            @Override
+            public void run(String... args) throws Exception {
+                Book book = new Book();
+                bookService.save(book);
+            }
+        };
+    }
+
```
Comment (okuzan) [inline src/main/java/com/github/onlinebookstore/OnlineBookStoreApplication.java:44]:
```suggestion
```

**Review by okuzan — CHANGES_REQUESTED** [review]

**Review by olekskov — APPROVED** [review]

**Pull request merged**